File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.kt
class Solution {
    fun minimumRecolors(blocks: String, k: Int): Int {
        var min = k
        var l = 0
        var black = 0
        var white = 0
        for (r in 0..blocks.length-1) {
            if (blocks[r] === "W".first()) {
                white++
            } else {
                black++
            }
            if (r - l + 1 === k) {
                min = Math.min(min, white)
                if (blocks[l] === "W".first()) {
                    white--
                } else {
                    black++
                }
                l++
            }
        }
        return min
    }
}

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.js
/**
 * @param {string} blocks
 * @param {number} k
 * @return {number}
 */
var minimumRecolors = function(blocks, k) {
    let l = 0
    let black = 0
    let white = 0
    let res = Infinity
    for (let r=0; r < blocks.length; r++) {
        if (blocks[r] === "W") {
            white++
        } else {
            black++
        }
        if (r - l + 1 === k) {
            res = Math.min(res, white)
            if (blocks[l] === "W") {
                white--
            } else {
                black++
            }
            l++
        }
    }
    return res
};